packages/near-membrane-base/src/intrinsics.ts

```diff
@@ -158,3 +160,13 @@ export function getFilteredEndowmentDescriptors(endowments: object): PropertyDes
     }
     return to;
 }
+
+export function getResolvedShapeDescriptors(
+    globalObjectShape: typeof globalThis,
+    endowments: object
+): PropertyDescriptorMap {
+    const globalObjectShapeDescriptors = getFilteredGlobalObjectShapeDescriptors(globalObjectShape);
```

Comment:
why not doing object.assign first, then one filter? not a big deal since they are supposed to be different sets.

Reply:
Because that will lose the non-enumerable own properties, which was part of the issue this change is solving: `'Promise' in Object.assign({}, window); // false`